src/Moq/Invocation.cs

```diff
@@ -89,6 +91,18 @@ public Exception Exception
 			}
 		}
 
+		public void ConvertResultToAwaitable(IAwaitableFactory awaitableFactory)
+		{
+			if (this.result is ExceptionResult r)
+			{
+				this.result = awaitableFactory.CreateFaulted(r.Exception);
+			}
```

Comment:
There may be circumstances where we may have to call the other `CreateFaulted(IEnumerable<Exception> exceptions)` overload instead (which, I'm assuming, is meant to cover `AggregateException`). I haven't yet looked into whether this is actually necessary. Perhaps it would be better to remove that other overload until that time?